pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -888,7 +888,7 @@ func ensureSystemCgroups(rootCgroupPath string, manager *fs.Manager) error {
 func isKernelPid(pid int) bool {
 	// Kernel threads have no associated executable.
 	_, err := os.Readlink(fmt.Sprintf("/proc/%d/exe", pid))
-	return err != nil
+	return err != nil && os.IsNotExist(err)
```

Comment:
Q: If it is a situation in which (for example) the file name is too long (i.e. an `ENAMETOOLONG` error), do we want to return that it isn't a kernel pid or do we want to return some other form of error message?

Reply:
It's hard to imagine a kernel PID would belong to a process spawned in a too long path so I guess not returning an error is fine here.